src/Client/BuzzHouse/Generator/SQLTypes.cpp

```diff
@@ -9,143 +8,1167 @@
+}
+
+JSONType::~JSONType()
+{
+    for (const auto & entry : subcols)
+    {
+        delete entry.subtype;
+    }
+}
+
+String Nullable::typeName(const bool escape) const
+{
+    return fmt::format("Nullable({})", subtype->typeName(escape));
+}
+
+String Nullable::MySQLtypeName(RandomGenerator & rg, const bool escape) const
+{
+    return subtype->MySQLtypeName(rg, escape);
+}
+
+String Nullable::PostgreSQLtypeName(RandomGenerator & rg, const bool escape) const
+{
+    return subtype->PostgreSQLtypeName(rg, escape);
+}
+
+String Nullable::SQLitetypeName(RandomGenerator & rg, const bool escape) const
+{
+    return subtype->SQLitetypeName(rg, escape);
+}
+
+SQLType * Nullable::typeDeepCopy() const
+{
+    return new Nullable(subtype->typeDeepCopy());
+}
+
+String Nullable::appendRandomRawValue(RandomGenerator & rg, StatementGenerator & gen) const
+{
+    return rg.nextMediumNumber() < 6 ? "NULL" : subtype->appendRandomRawValue(rg, gen);
+}
+
+String LowCardinality::typeName(const bool escape) const
+{
+    return fmt::format("LowCardinality({})", subtype->typeName(escape));
+}
+
+String LowCardinality::MySQLtypeName(RandomGenerator & rg, const bool escape) const
+{
+    return subtype->MySQLtypeName(rg, escape);
+}
+
+String LowCardinality::PostgreSQLtypeName(RandomGenerator & rg, const bool escape) const
+{
+    return subtype->PostgreSQLtypeName(rg, escape);
+}
+
+String LowCardinality::SQLitetypeName(RandomGenerator & rg, const bool escape) const
+{
+    return subtype->SQLitetypeName(rg, escape);
+}
+
+SQLType * LowCardinality::typeDeepCopy() const
+{
+    return new LowCardinality(subtype->typeDeepCopy());
+}
+
+String LowCardinality::appendRandomRawValue(RandomGenerator & rg, StatementGenerator & gen) const
+{
+    return subtype->appendRandomRawValue(rg, gen);
+}
+
+String GeoType::typeName(const bool) const
+{
+    return GeoTypes_Name(geotype);
+}
+
+String GeoType::MySQLtypeName(RandomGenerator &, const bool) const
+{
+    assert(0);
+    return "";
+}
+
+String GeoType::PostgreSQLtypeName(RandomGenerator &, const bool) const
+{
+    assert(0);
+    return "";
+}
+
+String GeoType::SQLitetypeName(RandomGenerator &, const bool) const
+{
+    assert(0);
+    return "";
+}
+
+SQLType * GeoType::typeDeepCopy() const
+{
+    return new GeoType(geotype);
+}
+
+String GeoType::appendRandomRawValue(RandomGenerator & rg, StatementGenerator &) const
+{
+    return strAppendGeoValue(rg, geotype);
+}
+
+String ArrayType::typeName(const bool escape) const
+{
+    return fmt::format("Array({})", subtype->typeName(escape));
+}
+
+String ArrayType::MySQLtypeName(RandomGenerator &, const bool) const
+{
+    assert(0);
+    return "";
+}
+
+String ArrayType::PostgreSQLtypeName(RandomGenerator & rg, const bool escape) const
+{
+    SQLType * nsubtype = subtype;
+    Nullable * nl = nullptr;
+    LowCardinality * lc = nullptr;
+
+    while (true)
+    {
+        if ((nl = dynamic_cast<Nullable *>(nsubtype)))
```

Comment:
You don't use `getTypeClass` because you really need to point to a valid object of the type, right?

Reply:
Yes, I need to print the type name of the subtype